src/progress.rs

```diff
@@ -170,6 +176,12 @@ where
             }
             Ok(StatusReport::NoOp) => {}
             Err(e) => {
+                poll_err_count += 1;
+
+                if poll_err_count < MAX_SUCCESSIVE_POLLING_ERRORS {
+                    continue;
+                }
```

Comment:
this is going to skip the sleep at the end of the loop, we probably still want to wait between retries

Reply:
good catch thanks!